pkg/relayer/proxy/proxy_test.go

```diff
@@ -1,3 +1,424 @@
+		},
+		{
+			desc: "Invalid relay supplier",
+
+			relayerProxyBehavior: []func(*testproxy.TestBehavior){
+				testproxy.WithRelayerProxyDependencies(supplierKeyName),
+				testproxy.WithRelayerProxiedServices(proxiedServices),
+				testproxy.WithDefaultActors(supplierKeyName, appPrivateKey),
+				// Missing session supplier
+				testproxy.WithDefaultSessionSupplier("", appPrivateKey),
+			},
+			inputScenario: sendRequestWithInvalidRelaySupplier,
+
+			expectedErrCode: -32000,
+			expectedErrMsg:  "invalid relayer proxy supplier",
+		},
+		{
+			desc: "Invalid signature",
+
+			relayerProxyBehavior: defaultBehavior,
+			inputScenario:        sendRequestWithInvalidSignature,
+
+			expectedErrCode: -32000,
+			expectedErrMsg:  "invalid ring signature",
+		},
+		{
+			desc: "Successful relay",
+
+			relayerProxyBehavior: defaultBehavior,
+			inputScenario:        sendRequestWithSuccessfulReply,
+
+			expectedErrCode: 0,
+			expectedErrMsg:  "",
+		},
+	}
+
+	ctx := context.TODO()
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(ctx)
+			test := testproxy.NewRelayerProxyTestBehavior(ctx, t, tt.relayerProxyBehavior...)
+
+			rp, err := proxy.NewRelayerProxy(
+				test.Deps,
+				proxy.WithSigningKeyName(supplierKeyName),
+				proxy.WithProxiedServicesEndpoints(proxiedServices),
+			)
+			require.NoError(t, err)
+
+			go rp.Start(ctx)
+			time.Sleep(100 * time.Millisecond)
+
+			errCode, errMsg := tt.inputScenario(t, test)
+			require.Equal(t, tt.expectedErrCode, errCode)
+			require.True(t, strings.HasPrefix(errMsg, tt.expectedErrMsg))
+
+			cancel()
+		})
+	}
+}
+
+func sendRequestWithUnparsableBody(
+	t *testing.T,
+	test *testproxy.TestBehavior,
+) (errorCode int32, errorMessage string) {
+	// Send non JSONRpc payload
+	reader := io.NopCloser(bytes.NewReader([]byte("invalid request")))
+
+	res, err := http.DefaultClient.Post(relayerProxyUrl, "application/json", reader)
+	require.NoError(t, err)
+	require.NotNil(t, res)
+
+	return testproxy.GetRelayResponseError(t, res)
+}
+
+func sendRequestWithMissingMeta(
+	t *testing.T,
+	test *testproxy.TestBehavior,
+) (errorCode int32, errorMessage string) {
+
+	req := &servicetypes.RelayRequest{
+		// RelayRequest is missing Metadata
+		Payload: validPayload,
+	}
+
+	return testproxy.MarshalAndSend(test, relayerProxyUrl, req)
+}
+
+func sendRequestWithMissingSignature(
+	t *testing.T,
+	test *testproxy.TestBehavior,
+) (errorCode int32, errorMessage string) {
+	req := testproxy.GenerateRelayRequest(test, appPrivateKey, "service1", 1, validPayload)
+	req.Meta.Signature = nil
+	return testproxy.MarshalAndSend(test, relayerProxyUrl, req)
+}
+
+func sendRequestWithInvalidRingSignature(
+	t *testing.T,
+	test *testproxy.TestBehavior,
+) (errorCode int32, errorMessage string) {
+	req := testproxy.GenerateRelayRequest(test, appPrivateKey, "service1", 1, validPayload)
+	req.Meta.Signature = []byte("invalid ring signature")
```

Comment:
NIT: Remove `ring` from the function name, error message and test case. It should be agnostic to the type of signature under the hood.